src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1409,6 +1420,97 @@ def disable_features(cmd, client, resource_group_name, cluster_name, features, k
     return str.format(consts.Successfully_Disabled_Features, features, connected_cluster.name)
 
 
+def troubleshoot(cmd, client, resource_group_name, cluster_name, kube_config=None, kube_context=None, location=None, storage_account=None,
+                 sas_token=None, output_file=os.path.join(os.path.expanduser('~'), '.azure', 'az_connectedk8s_troubleshoot_output.tar.gz')):
+    colorama.init()
+    print(f"{colorama.Fore.YELLOW}Troubleshooting the ConnectedCluster for possible issues...")
+    utils.check_connectivity()  # Checks internet connectivity
+    troubleshoot_log_path = os.path.join(os.path.expanduser('~'), '.azure', 'connected8s_troubleshoot.log')
+    utils.setup_logger('connectedk8s_troubleshoot', troubleshoot_log_path)
+    tr_logger = logging.getLogger('connectedk8s_troubleshoot')  # logger for troubleshooting, onto a log file
+
+    # Send cloud information to telemetry
+    send_cloud_telemetry(cmd)
+
+    # setting kubeconfig
+    kube_config = set_kube_config(kube_config)
+
+    # Loading the kubeconfig file in kubernetes client configuration
+    load_kube_config(kube_config, kube_context)
+    configuration = kube_client.Configuration()
+    validate_release_namespace(client, cluster_name, resource_group_name, configuration, kube_config, kube_context)
```

Comment:
Why are we doing this in connectedk8s troubleshoot?  We already execute this function in 'connectedk8s connect' or 'enable/disable features'. User will end up getting the same error message. I don't see any use case here, is there?

Reply:
Yes, these are in those commands too. 

This is just to complete the full circle of sanity checks in the troubleshoot also, so that these things can be verified, before running into periscope checks. 
Also, this answers your https://github.com/Azure/azure-cli-extensions/pull/3392#discussion_r669581316